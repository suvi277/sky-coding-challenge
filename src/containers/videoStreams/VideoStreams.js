import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchStreamData } from 'actions';
import StreamCard from 'components/streamCard/StreamCard';
import Placeholder from 'components/placeholder/Placeholder';
import Aux from 'components/hoc/Aux';
import FadeIn from 'react-fade-in';

export class VideoStreams extends Component {
	componentDidMount() {
		this.props.fetchStreamData();
	}

	render() {
		const { isLoaded, content } = this.props.streamData;
		return (
			<Aux>
				{!isLoaded ? (
					<div className="d-flex flex-wrap">{[ ...Array(12) ].map((x, i) => <Placeholder key={i} />)}</div>
				) : (
					<FadeIn>
						<div className="d-flex flex-wrap">
							{content.map((item, i) => <StreamCard key={i} {...item} />)}
						</div>
					</FadeIn>
				)}
			</Aux>
		);
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		fetchStreamData: () => dispatch(fetchStreamData())
	};
};

export default connect(({ streamData }) => ({ streamData }), mapDispatchToProps)(VideoStreams);
