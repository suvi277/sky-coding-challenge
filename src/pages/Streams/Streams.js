import React from 'react';
import MainLayout from 'components/hoc/Layout';
import PageHeading from 'components/pageHeading/PageHeading';
import VideoStreams from 'containers/videoStreams/VideoStreams';

const Home = () => {
	return (
		<MainLayout>
			<PageHeading title="Video Streams" />
			<VideoStreams />
		</MainLayout>
	);
};
export default Home;