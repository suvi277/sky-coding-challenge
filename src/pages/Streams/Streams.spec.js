import React from 'react';
import { shallow } from 'enzyme';
import Streams from './Streams';
import VideoStreams from 'containers/videoStreams/VideoStreams';

describe('Streams Component', () => {
	const wrapper = shallow(<Streams />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should exist MainLayout Component', () => {
		expect(wrapper.find('MainLayout')).toHaveLength(1);
	});

	it('should exist PageHeading Component', () => {
		expect(wrapper.find('PageHeading')).toHaveLength(1);
	});

	it('should exist VideoStreams Component', () => {
		expect(wrapper.exists(VideoStreams)).toBeTruthy();
	});
});
