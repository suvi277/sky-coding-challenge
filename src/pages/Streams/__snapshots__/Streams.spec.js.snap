// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Streams Component renders as expected 1`] = `
<MainLayout>
  <PageHeading
    title="Video Streams"
  />
  <Connect(VideoStreams) />
</MainLayout>
`;
